courseware/api_test.py

```diff
@@ -148,51 +149,62 @@ def test_create_edx_auth_token(settings, user):
 
 @responses.activate
 @freeze_time("2019-03-24 11:50:36")
-def test_refresh_edx_api_auth(settings):
-    """Tests refresh_edx_api_auth makes the expected incantations to create a OpenEdxApiAuth"""
-    auth = OpenEdxApiAuthFactory.create(expired=True)
+@pytest.mark.parametrize("expired", [True, False])
+def test_get_valid_edx_api_auth(settings, expired):
+    """Tests get_valid_edx_api_auth makes the expected incantations to create a OpenEdxApiAuth"""
+    auth = OpenEdxApiAuthFactory.create(expired=expired)
     refresh_token = "abc123"
     access_token = "def456"
-    responses.add(
-        responses.POST,
-        f"{settings.OPENEDX_API_BASE_URL}/oauth2/access_token",
-        json=dict(
-            refresh_token=refresh_token, access_token=access_token, expires_in=3600
-        ),
-        status=status.HTTP_200_OK,
-    )
-
-    refresh_edx_api_auth(auth.user)
-
-    assert len(responses.calls) == 1
-    assert dict(parse_qsl(responses.calls[0].request.body)) == dict(
-        refresh_token=auth.refresh_token,
-        grant_type="refresh_token",
-        client_id=settings.OPENEDX_API_CLIENT_ID,
-        client_secret=settings.OPENEDX_API_CLIENT_SECRET,
-    )
-
-    auth.refresh_from_db()
-
-    assert auth.refresh_token == refresh_token
-    assert auth.access_token == access_token
-    # plus expires_in, minutes 10 seconds
-    assert auth.access_token_expires_on == now_in_utc() + timedelta(
-        minutes=59, seconds=50
-    )
+    if expired:
```

Comment:
There's enough forked logic here that I think it makes more sense to just split this into 2 different test cases (`test_get_valid_edx_api_auth_unexpired` and `test_get_valid_edx_api_auth_expired`, et al)